msgpack-core/src/main/java/org/msgpack/core/MessageUnpacker.java

```diff
@@ -1005,9 +1004,10 @@ public String unpackString() throws IOException {
                         sb.append(decodeBuffer);
 
                         decodeBuffer.clear();
-                        cursor += readLen;
-                        consume(readLen);
                     }
+
+                    cursor += readLen;
```

Comment:
I think there is no guarantee that the read length from the input always matches `Math.min(buffer.size() - position, strLen-cursor)`  when OVERFLOW (output buffer has insufficient space) happens.  In this case, the input data is partially read. That is why I compute `readLen` as `bb.limit() - bb.remaining()`.